packages/synapse-guest-module/synapse_guest_module/guest_module.py

```diff
@@ -138,6 +141,34 @@ async def callback_user_may_invite(
         user_is_guest = inviter.startswith("@" + self._config.user_id_prefix)
         return not user_is_guest
 
+    async def callback_user_may_join_room(
+        self,
+        user_id: str,
+        room_id: str,
+        is_invited: bool
+    ) -> str:
+        """Returns whether this user is allowed to join a room. Guest users
+        should only be able to do that if the room is Ask to Join (knock).
+        """
+        user_is_guest = user_id.startswith("@" + self._config.user_id_prefix)
```

Comment:
We should eventually refactor this check into a method

Reply:
didn't bother to, as all the other callback's aren't doing so...